src/optim_tests/multivariate/quad_transforms.jl

```diff
@@ -0,0 +1,141 @@
+                        initial_x,
+                        x0,
+                        zero(T),
+                        true,
+                        true,
+                        MatVecHolder(mat, x0))
+end
+
+examples["Quadratic Diagonal"] = _quadraticproblem(100)
+
+#######################
+# Paraboloid. Similar to Rosenbrock
+#######################
+
+immutable ParaboloidStruct{T, Tm <: AbstractArray{T,2},
+                           Tv <: AbstractArray{T}} <: Any where T<:Number
+
+    mat::Tm
+    vec::Tv
+    xt::Tv
+    alpha::T
+end
+
+function paraboloid(x::AbstractArray, param::ParaboloidStruct)
+    mat = param.mat
+    xt = param.xt
+    @. xt = x - param.vec
+    xt[2:end] .-= param.alpha*xt[1]^2
+
+    return 0.5*vecdot(xt, mat*xt)
+end
+
+function paraboloid_gradient!(storage::AbstractArray, x::AbstractArray, param::ParaboloidStruct)
+    mat = param.mat
+    xt = param.xt
+
+    @. xt = x - param.vec
+    xt[2:end] .-= param.alpha*xt[1]^2
+
+    storage .= mat*xt
+    storage[1] -= 2.0*param.alpha*xt[1]*sum(storage[2:end])
+end
+
+
+function paraboloid_fun_gradient!(storage::AbstractArray, x::AbstractArray, param::ParaboloidStruct)
+    mat = param.mat
+    xt = param.xt
+
+    @. xt = x - param.vec
+    xt[2:end] .-= param.alpha*xt[1]^2
+
+    storage .= mat*xt
+    storage[1] -= 2.0*param.alpha*xt[1]*sum(storage[2:end])
+
+    return 0.5*vecdot(xt, mat*xt)
+end
+
+function paraboloid_hessian!(storage,x,param)
+    error("Hessian not implemented for Paraboloid")
+end
+
+function _paraboloidproblem(N::Int; mat::AbstractArray{T,2} = spdiagm(float(1:N)),
+                            x0::AbstractVector{T} = ones(N),
+                            initial_x::AbstractVector{T} = zeros(N),
+                            alpha::T = 10.0,
+                            name::AbstractString = "Paraboloid Diagonal ($N)") where T
+    OptimizationProblem(name,
+                        paraboloid,
+                        paraboloid_gradient!,
+                        paraboloid_fun_gradient!,
+                        paraboloid_hessian!,
+                        initial_x,
+                        x0,
+                        zero(T),
+                        true,
+                        false,
+                        ParaboloidStruct(mat, x0, similar(x0), alpha))
+end
+
+examples["Paraboloid Diagonal"] = _paraboloidproblem(100)
+
+function _randommatrix(N::Int, scaling::Bool=true)
+    F = qrfact(randn(N,N))
+    if scaling
+        retval = F[:Q]'*spdiagm(float(1:N))*F[:Q]
+    else
+        retval = F[:Q]'*F[:Q]
+    end
+    retval
+end
+
+srand(0) # TODO: is it dodgy to set the random seed in a package?
+examples["Paraboloid Random matrix"] = _paraboloidproblem(100;
+                                                          name = "Paraboloid Random Matrix (100)",
+                                                          mat = _randommatrix(100))
+srand()  # TODO: is it fine if we reset the seed after?
```

Comment:
Is this (and L137) bad practice? Are there situations where calling `srand` via `using` is problematic?

Reply:
Does it affect code outside of the module's scope?

Reply:
Hmm, from trying this out, I'm not really sure how comfortable I am with this. I am a bit worried that even if it's only us who are using this, we might screw this up when using random numbers to benchmark or something similar. Can we not just store the matrix in explicit form? I know it's a bit... but...